lib/plugin.js

```diff
@@ -94,20 +94,16 @@ ManifestPlugin.prototype.apply = function(compiler) {
 
     var json = JSON.stringify(cache, null, 2);
 
-    compilation.assets[outputName] = {
-      source: function() {
-        return json;
-      },
-      size: function() {
-        return json.length;
-      }
-    };
-
-    if (this.opts.writeToFileEmit) {
-      var outputFolder = compilation.options.output.path;
-      var outputFile = path.join(outputFolder, this.opts.fileName);
+    var outputFolder = compilation.options.output.path;
+    var outputFile = path.join(outputFolder, this.opts.fileName);
 
+    const shouldUseNodeFs = compiler.outputFileSystem.constructor.name !== 'MemoryFileSystem' ||
+                                      this.opts.writeToFileEmit;
```

Comment:
this indentation looks broken, I think we should stick to 2 spaces everywhere, also I think it is little bit more readable if we put `||` on the new line:
```js
const shouldUseNodeFs = compiler.outputFileSystem.constructor.name !== 'MemoryFileSystem'
  || this.opts.writeToFileEmit;
```